### ISSUE TITLE
Impossible to rerun the pageobject example tests in "--watch" mode

### ISSUE THREAD

**Issue by omaiboroda** [webdriverio/webdriverio#2596]
## The problem

After the second rerun in "--watch" mode, the "elements" cannot be found.

## Environment

* WebdriverIO version: 4.10.2
* Node.js version: v8.9.4
* Wdio testrunner:
* Synchronous mode:

## Details

Elements are not found after tests rerun in "--watch" mode. Steps to reproduce for _pageobject_ example:

1. Run the _pageobject_ example in watch mode: `./bin/wdio examples/pageobject/wdio.conf.js --watch`
2. Save _examples/pageobject/specs/form.spec.js_ to trigger the test run
3. Tests are passing
4. Resave _form.spec.js_ once more to trigger the test run for second time
5. Tests are not passing, `("#password")` element is not found.

This strange behavior is related to lastPromise/lastResult - objects are is not cleared(should them?), and "#password" element is looked up in scope of the previous element
POST "_/wd/hub/session/bfd14c20c7a8206a577fd3c9997b6f98/element/0.06965603720256963-1/element_", 
rather then 
_POST "/wd/hub/session/bfd14c20c7a8206a577fd3c9997b6f98/element"_

Any ideas on it?

## Link to Selenium/WebdriverIO logs

[Logs with error here](https://gist.github.com/omaiboroda/edba676edb9cd95a5e40e73233c5cd9e) 

_UPD:_

It seems, this is related with pageobject approach and command chaining - the problem persists with 
`FormPage.password.setValue('SuperSecretPassword');`
but doesn't with 
`browser.setValue('#password', 'SuperSecretPassword')`

**Comment by presidenten** [comment]
Found a way around the issue .. which unfortunately is a bit slower since it needs to restart the browser everytime the tests are run.

npm i onchange --dev

then in package.json scripts:
```
  "test:dev": "onchange '<test-dir or blob here>' -d 10 -- npm run test"
```

**Comment by christian-bromann** [comment]
implemented as part of #2857
